test/test.js

```diff
@@ -112,41 +112,49 @@ describe('Page', function() {
     state.browser = null;
   });
 
-  beforeEach(async(state, test) => {
-    state.page = await state.browser.newPage();
-    const rl = require('readline').createInterface({input: state.browser.process().stderr});
-    test.output = '';
-    rl.on('line', onLine);
-    state.tearDown = () => {
-      rl.removeListener('line', onLine);
-      rl.close();
-    };
-    function onLine(line) {
-      test.output += line + '\n';
-    }
-  });
-
-  afterEach(async state => {
-    state.tearDown();
-    await state.page.close();
-    state.page = null;
+  describe('Page', function() {
+    beforeEach(async(state, test) => {
+      state.context = await state.browser.createIncognitoBrowserContext();
+      state.page = await state.context.newPage();
+      const rl = require('readline').createInterface({input: state.browser.process().stderr});
```

Comment:
Can you move the stderr code into its own beforeEach in the `Browser` block? Otherwise the browser context tests lose out on stderr recording.